File: 231213/모이자/gather.js
const fs = require('fs');
const input = () => fs.readFileSync('/dev/stdin').toString().split('\n');

let [N, arr] = input()

N = Number(N)
arr = arr.split(' ').map(element => Number(element));

let minDistance = Infinity;

for(let i = 0; i < arr.length; i++){
    tempSum = 0;

    for(let j = 0; j < arr.length; j++){
        tempSum += Math.abs(i - j) * arr[j];
    }

    minDistance = Math.min(minDistance, tempSum);
}

console.log(minDistance)

File: 231213/모이자/gather.py
# 합이 최소
import sys

N = int(input())
arr = list(map(int, input().split()))

minSum = sys.maxsize

for i in range(N):
    tempSum = 0

    for j in range(N):
        tempSum += abs(i - j) * arr[j]

    minSum = min(minSum, tempSum)

print(minSum)